From e4ea51b9afb120b0d80db9ad82020a3246675a4e Mon Sep 17 00:00:00 2001
From: Dominique Leuenberger <dimstar@opensuse.org>
Date: Fri, 12 Feb 2021 10:47:22 +0100
Subject: [PATCH] Do not inject 'no-separator' into window decoration

Gtk3::Dialog->new was called with the parameter "no-separator"
which is invalid for Gtk3 as the error message states (it was valid in
Gtk2)

This is exposed by perl-Gtk3 0.038, which handled these flags in a buggy
way. The fix in perl-Gtk3 exposed the issue in clamtk.
---
 lib/Analysis.pm | 2 +-
 lib/History.pm  | 2 +-
 lib/Scan.pm     | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/Analysis.pm b/lib/Analysis.pm
index 06a1bf7..6a82933 100644
--- a/lib/Analysis.pm
+++ b/lib/Analysis.pm
@@ -48,7 +48,7 @@ sub show_window {
     ( undef, $from_scan, $parent ) = @_;
     $window = Gtk3::Dialog->new(
         undef, $parent,
-        [   qw| modal destroy-with-parent no-separator
+        [   qw| modal destroy-with-parent 
                 use-header-bar |
         ],
     );
diff --git a/lib/History.pm b/lib/History.pm
index f84aa97..b5906be 100644
--- a/lib/History.pm
+++ b/lib/History.pm
@@ -148,7 +148,7 @@ sub view_history {
 
     my $win = Gtk3::Dialog->new(
         sprintf( _( 'Viewing %s' ), $basename ),
-        undef, [ qw| modal destroy-with-parent no-separator | ],
+        undef, [ qw| modal destroy-with-parent | ],
     );
     $win->signal_connect( destroy => sub { $win->destroy; 1 } );
     $win->set_default_size( 800, 350 );
diff --git a/lib/Scan.pm b/lib/Scan.pm
index 0de14df..b1a32fc 100644
--- a/lib/Scan.pm
+++ b/lib/Scan.pm
@@ -90,7 +90,7 @@ sub filter {
     # Begin popup scanning
     $window = Gtk3::Dialog->new(
         undef, undef,
-        [   qw| modal destroy-with-parent no-separator
+        [   qw| modal destroy-with-parent 
                 use-header-bar |
         ],
     );
-- 
2.30.0

